ui/src/dropdown.tsx

```diff
@@ -109,24 +110,19 @@ const
         // HACK: Push clears args so run it after useEffect sets them due to model.value change.
         if (trigger) setTimeout(() => wave.push(), 0)
       },
-      selectAll = () => {
+      selectAll = (checked = true) => () => {
         if (!selection) return
+        if (checked) selection.clear()
 
-        selection.clear()
-        options.forEach(o => { if (!o.disabled) selection.add(o.key as S) })
-
-        const selectionArr = Array.from(selection)
-        setMultiValues(selectionArr)
-        wave.args[name] = selectionArr
-
-        onChange()
-      },
-      deselectAll = () => {
-        if (!selection) return
+        options.forEach(({ disabled, key }) => {
+          checked
+            ? !disabled || values?.includes(key as S) ? selection.add(key as S) : null
```

Comment:
```js
? !disabled || values?.includes(key as S) ? selection.add(key as S) : null
```

Clearing whole selection and then putting things back seems odd. Isn't there a more elegant way?

Reply:
Considering we want to keep the Javascript Set for storing the selection in the `BaseDropdown`, there is an option of adding without clearing, but in this case selected choices are not in the original order so it is inconsistent with a `DialogDropdownMulti` where selected choices are in the original order when using Select all.

https://github.com/h2oai/wave/assets/23740173/4dceadbc-0822-427f-b442-4f2be6a0a732

Another option would be to go without Set.

Reply:
> but in this case selected choices are not in the original order

What is the definition of "original" order?

> Another option would be to go without Set.

Yes, at first glance, Set seems redundant here.

> it is inconsistent with a DialogDropdownMulti

Both dropdown types must behave exactly the same, achieved by the most straightforward code possible.

Reply:
> What is the definition of "original" order?

As present in `choices` model value.